### PR TITLE
added logs for when miner state changes

### PR DESCRIPTION
**Pull request by laudiacay**
This PR adds logs for when miner state balances change as requested in #1500. There were some spots where i wasn’t sure if the functions in miner_state.go were returning the right value, so i flagged those, and also found one spot where the wrong thing was being logged and fixed it.

This PR came from my IPad so sorry in advance if there’s anything wonky... tried to check diffs over carefully

### CODE REVIEW

File: actors/builtin/miner/miner_actor.go [hunk 1]
```diff
@@ -2237,9 +2254,11 @@ func handleProvingDeadline(rt Runtime) {
 		{
 			depositToBurn, err := st.CleanUpExpiredPreCommits(store, currEpoch)
 			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to expire pre-committed sectors")
+			rt.Log(rtt.INFO, "subtracted burnt deposits from precommit deposit while cleaning up expired precommits. amount burnt: %v, new total: %v", depositToBurn, st.PreCommitDeposits)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2257]:
```suggestion
			rt.Log(rtt.INFO, "subtracted burnt deposits from precommit deposit while cleaning up expired precommits. amount burnt: %v, new total: %v", depositToBurn, st.PreCommitDeposits)
```
```suggestion
			rt.Log(rtt.INFO, "subtracted burnt deposits from precommit deposit while cleaning up expired precommits, amount burnt: %v, new total: %v", depositToBurn, st.PreCommitDeposits)
```

File: actors/builtin/miner/miner_actor.go [hunk 2]
```diff
@@ -2169,14 +2183,17 @@ func processEarlyTerminations(rt Runtime) (more bool) {
...
 		// Remove pledge requirement.
 		err = st.AddInitialPledge(totalInitialPledge.Neg())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add initial pledge %v", totalInitialPledge.Neg())
+		rt.Log(rtt.INFO, "added initial pledge. amt added: %v, new total: %v", totalInitialPledge.Neg(), st.InitialPledge)
 		pledgeDelta = big.Sub(pledgeDelta, totalInitialPledge)
 
 		// Use unlocked pledge to pay down outstanding fee debt
 		penaltyFromVesting, penaltyFromBalance, err := st.RepayPartialDebtInPriorityOrder(store, rt.CurrEpoch(), rt.CurrentBalance())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pay penalty")
+		rt.Log(rtt.INFO, "paid fee debt. amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2196]:
```suggestion
		rt.Log(rtt.INFO, "paid fee debt, amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 3]
```diff
@@ -2169,14 +2183,17 @@ func processEarlyTerminations(rt Runtime) (more bool) {
 		err = st.ApplyPenalty(penalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
 
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", penalty, st.FeeDebt)
 		// Remove pledge requirement.
 		err = st.AddInitialPledge(totalInitialPledge.Neg())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add initial pledge %v", totalInitialPledge.Neg())
+		rt.Log(rtt.INFO, "added initial pledge. amt added: %v, new total: %v", totalInitialPledge.Neg(), st.InitialPledge)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2190]:
```suggestion
		rt.Log(rtt.INFO, "added initial pledge, amt added: %v, new total: %v", totalInitialPledge.Neg(), st.InitialPledge)
```

File: actors/builtin/miner/miner_actor.go [hunk 4]
```diff
@@ -2169,14 +2183,17 @@ func processEarlyTerminations(rt Runtime) (more bool) {
 		err = st.ApplyPenalty(penalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
 
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", penalty, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2186]:
```suggestion
		rt.Log(rtt.INFO, "added penalty to fee debt, amount added: %v, new total fee debt: %v", penalty, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 5]
```diff
@@ -2058,6 +2071,7 @@ func (a Actor) RepayDebt(rt Runtime, _ *abi.EmptyValue) *abi.EmptyValue {
 		// Repay as much fee debt as possible.
 		fromVesting, fromBalance, err = st.RepayPartialDebtInPriorityOrder(adt.AsStore(rt), rt.CurrEpoch(), rt.CurrentBalance())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock fee debt")
+		rt.Log(rtt.INFO, "paid fee debt. amount from vesting table: %v, amount from balance: %v, new fee debt: %v", fromVesting, fromBalance, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2074]:
```suggestion
		rt.Log(rtt.INFO, "paid fee debt, amount from vesting table: %v, amount from balance: %v, new fee debt: %v", fromVesting, fromBalance, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 6]
```diff
@@ -1947,9 +1956,11 @@ func (a Actor) ReportConsensusFault(rt Runtime, params *ReportConsensusFaultPara
 		err := st.ApplyPenalty(faultPenalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
 
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", faultPenalty, st.FeeDebt)
 		// Pay penalty
 		penaltyFromVesting, penaltyFromBalance, err := st.RepayPartialDebtInPriorityOrder(adt.AsStore(rt), currEpoch, rt.CurrentBalance())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pay fees")
+		rt.Log(rtt.INFO, "paid fee debt. amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1963]:
```suggestion
		rt.Log(rtt.INFO, "paid fee debt, amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 7]
```diff
@@ -1947,9 +1956,11 @@ func (a Actor) ReportConsensusFault(rt Runtime, params *ReportConsensusFaultPara
 		err := st.ApplyPenalty(faultPenalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
 
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", faultPenalty, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1959]:
```suggestion
		rt.Log(rtt.INFO, "added penalty to fee debt, amount added: %v, new total fee debt: %v", faultPenalty, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 8]
```diff
@@ -1869,17 +1874,21 @@ func (a Actor) ApplyRewards(rt Runtime, params *builtin.ApplyRewardParams) *abi.
...
 		// If the miner incurred block mining penalties charge these to miner's fee debt
 		err = st.ApplyPenalty(params.Penalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", params.Penalty, st.FeeDebt)
 		// Attempt to repay all fee debt in this call. In most cases the miner will have enough
 		// funds in the *reward alone* to cover the penalty. In the rare case a miner incurs more
 		// penalty than it can pay for with reward and existing funds, it will go into fee debt.
 		penaltyFromVesting, penaltyFromBalance, err := st.RepayPartialDebtInPriorityOrder(store, rt.CurrEpoch(), rt.CurrentBalance())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to repay penalty")
+		rt.Log(rtt.INFO, "paid fee debt. amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1891]:
```suggestion
		rt.Log(rtt.INFO, "paid fee debt, amount from vesting table: %v, amount from balance: %v, new fee debt: %v", penaltyFromVesting, penaltyFromBalance, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 9]
```diff
@@ -1869,17 +1874,21 @@ func (a Actor) ApplyRewards(rt Runtime, params *builtin.ApplyRewardParams) *abi.
...
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to lock funds in vesting table")
+		/// XXX am unsure if this is correct! should i be logging amountUnlocked, or VestingSum from addLockedFunds?? could the return value be a bug? confused about what's going on.
+		rt.Log(rtt.INFO, "unlocked some funds. amount unlocked: %v, new amount locked: %v.", newlyVested, st.LockedFunds)
 		pledgeDeltaTotal = big.Sub(pledgeDeltaTotal, newlyVested)
 		pledgeDeltaTotal = big.Add(pledgeDeltaTotal, rewardToLock)
 
 		// If the miner incurred block mining penalties charge these to miner's fee debt
 		err = st.ApplyPenalty(params.Penalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
+		rt.Log(rtt.INFO, "added penalty to fee debt. about added: %v, new total fee debt: %v", params.Penalty, st.FeeDebt)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1885]:
```suggestion
		rt.Log(rtt.INFO, "added penalty to fee debt, amount added: %v, new total fee debt: %v", params.Penalty, st.FeeDebt)
```

File: actors/builtin/miner/miner_actor.go [hunk 10]
```diff
@@ -1869,17 +1874,21 @@ func (a Actor) ApplyRewards(rt Runtime, params *builtin.ApplyRewardParams) *abi.
 
 		newlyVested, err := st.AddLockedFunds(store, rt.CurrEpoch(), rewardToLock, lockedRewardVestingSpec)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to lock funds in vesting table")
+		/// XXX am unsure if this is correct! should i be logging amountUnlocked, or VestingSum from addLockedFunds?? could the return value be a bug? confused about what's going on.
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1877]: I think the return value is correct -- As a side effect of adding locked funds,  some vested funds could be unlocked. The return is just communicating this update.

I don't think this is worth logging at all.

File: actors/builtin/miner/miner_actor.go [hunk 11]
```diff
@@ -2016,6 +2027,8 @@ func (a Actor) WithdrawBalance(rt Runtime, params *WithdrawBalanceParams) *abi.T
 		if err != nil {
 			rt.Abortf(exitcode.ErrIllegalState, "failed to vest fund: %v", err)
 		}
+		/// XXX unsure if returning/logging the right thing here, again
+		rt.Log(rtt.INFO, "unlocked vested funds. amount unlocked: %v, new total locked: %v", newlyVested, st.LockedFunds)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2031]: Also not worth logging imo

**Review by arajasek — CHANGES_REQUESTED** [review]
Some remarks:

- i don't think periods in logs is standard go practice...suggested changing those to commas
- would consider dropping most of these logs to DEBUG, but defer to @jennijuju 
- `about` and `amount` are different words, iPad lady

File: actors/builtin/miner/miner_state.go [hunk 12]
```diff
@@ -1013,7 +1013,7 @@ func (st *State) CheckBalanceInvariants(balance abi.TokenAmount) error {
 		return xerrors.Errorf("initial pledge is negative: %v", st.InitialPledge)
 	}
 	if st.FeeDebt.LessThan(big.Zero()) {
-		return xerrors.Errorf("fee debt is negative: %v", st.InitialPledge)
+		return xerrors.Errorf("fee debt is negative: %v", st.FeeDebt)
```
[the remark this answers is not in the document]
Comment (laudiacay) [reply actors/builtin/miner/miner_state.go:1016]: ?

**Comment by laudiacay** [comment]
Closes #1500

File: actors/builtin/miner/miner_actor.go [hunk 13]
```diff
@@ -1204,6 +1207,7 @@ func confirmSectorProofsValid(rt Runtime, preCommits []*SectorPreCommitOnChainIn
 		// Unlock deposit for successful proofs, make it available for lock-up as initial pledge.
 		err = st.AddPreCommitDeposit(depositToUnlock.Neg())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add pre-commit deposit %v", depositToUnlock.Neg())
+		rt.Log(rtt.DEBUG, "added precommit deposit, amt added: %v, new total: %v", depositToUnlock.Neg(), st.PreCommitDeposits)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:1210]: this (and the `require` above it) are actually unlocking deposit, not adding any

File: actors/builtin/miner/miner_actor.go [hunk 14]
```diff
@@ -2169,14 +2180,17 @@ func processEarlyTerminations(rt Runtime) (more bool) {
 		err = st.ApplyPenalty(penalty)
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
 
+		rt.Log(rtt.DEBUG, "added penalty to fee debt, amount added: %v, new total fee debt: %v", penalty, st.FeeDebt)
 		// Remove pledge requirement.
 		err = st.AddInitialPledge(totalInitialPledge.Neg())
 		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to add initial pledge %v", totalInitialPledge.Neg())
+		rt.Log(rtt.DEBUG, "added initial pledge, amt added: %v, new total: %v", totalInitialPledge.Neg(), st.InitialPledge)
```
Comment (arajasek) [inline actors/builtin/miner/miner_actor.go:2187]: also actually unlocking

**Review by arajasek — APPROVED** [review]

**Pull request merged**